File: lwjgl/src/main/java/com/darkraha/opengldemoj/gl/GlObject.java
package com.darkraha.opengldemoj.gl;

import com.darkraha.opengldemoj.gl.modelling.GlModel;
import org.joml.Matrix4f;

public class GlObject {
    public GlModel model;
    public GlTexture texture; // primary texture
    public GlTexture normalTexture;
    public Matrix4f transforms;
    public GlTexture []extraTextures;
}

File: lwjglkt/src/main/kotlin/com/darkraha/opengldemokt/gl/GlObject.kt
package com.darkraha.opengldemokt.gl


import com.darkraha.opengldemokt.gl.modelling.GlModel
import org.joml.Matrix4f

class GlObject {
    var model: GlModel? = null
    var texture: GlTexture? = null
    var normalTexture: GlTexture? = null
    var transforms: Matrix4f? = null
    var extraTextures: Array<GlTexture>?=null
}
